crates/ruff_python_semantic/src/model.rs

```diff
@@ -336,7 +347,12 @@ impl<'a> SemanticModel<'a> {
                     //
                     // The `x` in `print(x)` should be treated as unresolved.
                     BindingKind::Deletion | BindingKind::UnboundException(None) => {
-                        return ResolvedRead::UnboundLocal(binding_id)
+                        self.unresolved_references.push(
+                            range,
+                            self.exceptions(),
+                            UnresolvedReferenceFlags::empty(),
+                        );
```

Comment:
Don't love that we're pushing the unresolved reference in this method... but it's consistent with how resolved references are treated.